pkg/dies/cartographer/v1alpha1/clustersupply_chain.go

```diff
@@ -0,0 +1,40 @@
+package v1alpha1
+
+import (
+	diemetav1 "dies.dev/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	cartov1alpha1 "github.com/vmware-tanzu/apps-cli-plugin/pkg/apis/cartographer/v1alpha1"
+)
+
+// +die:object=true
+type _ = cartov1alpha1.ClusterSupplyChain
+
+// +die
+type _ = cartov1alpha1.SupplyChainSpec
+
+// +die
+type _ = cartov1alpha1.SupplyChainStatus
+
+func (d *SupplyChainStatusDie) ConditionsDie(conditions ...*diemetav1.ConditionDie) *SupplyChainStatusDie {
+	return d.DieStamp(func(r *cartov1alpha1.SupplyChainStatus) {
+		r.Conditions = make([]metav1.Condition, len(conditions))
+		for i := range conditions {
+			r.Conditions[i] = conditions[i].DieRelease()
+		}
+	})
+}
+
+// StatusDie stamps the resource's Status field with a mutable die.
+func (d *ClusterSupplyChainDie) StatusDie(fn func(d *SupplyChainStatusDie)) *ClusterSupplyChainDie {
+	return d.DieStamp(func(r *cartov1alpha1.ClusterSupplyChain) {
+		d := SupplyChainStatusBlank.DieImmutable(false).DieFeed(r.Status)
+		fn(d)
+		r.Status = d.DieRelease()
+	})
+}
```

Comment:
this isn't documented. When you have spec/status structs that don't follow the naming conventions, you can override the default values. This will generate the SpecDie and StatusDie methods you don't have to define them manually.

```suggestion
// +die:object=true,spec=SupplyChainSpec,status=SupplyChainStatus
type _ = cartov1alpha1.ClusterSupplyChain

// +die
type _ = cartov1alpha1.SupplyChainSpec

// +die
type _ = cartov1alpha1.SupplyChainStatus

func (d *SupplyChainStatusDie) ConditionsDie(conditions ...*diemetav1.ConditionDie) *SupplyChainStatusDie {
	return d.DieStamp(func(r *cartov1alpha1.SupplyChainStatus) {
		r.Conditions = make([]metav1.Condition, len(conditions))
		for i := range conditions {
			r.Conditions[i] = conditions[i].DieRelease()
		}
	})
}
```

Reply:
I was planning to ask you about this. Thanks for the suggestion